log/include/gz/utils/log/SplitSink.hh

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GZ_UTILS_LOG_SPLITSINK_HH_
+#define GZ_UTILS_LOG_SPLITSINK_HH_
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <spdlog/details/log_msg.h>
+#include <spdlog/details/null_mutex.h>
+#include <spdlog/pattern_formatter.h>
+#include <spdlog/sinks/base_sink.h>
+#include <spdlog/sinks/ringbuffer_sink.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+
+namespace gz::utils::log
+{
+/// \brief Logging sink for spdlog that logs in Gazebo-conventions.
+///
+/// This will route messages with severity (warn, err, critical) to stderr,
+/// and all other levels (info, debug, trace) to stdout.
+template<typename Mutex>
```

Comment:
Do we expect to use other kinds of mutexes? i.e, do we need to provide this template parameter? Unless, there's a good reason, my preference would be to make this a regular class and follow our common practice of using an ImplPtr so we can make changes to it down the road.

Reply:
@mjcarroll , do you have an opinion here?

Reply:
I don't have a strong opinion, I was following the pattern of spdlog upstream, as they provide st/mt logger sink implementations.  I think for gazebo, the mt logger (the one that requires mutexes) is probably always what we would want, so we could safely drop the template here.